File: backend/app/schemas/stats.py
from pydantic import BaseModel


class LabelTotal(BaseModel):
    label: str
    total_seconds: int


class TasksPerDay(BaseModel):
    date: str
    count: int


class Distribution(BaseModel):
    label: str
    percentage: float


class DailySeries(BaseModel):
    date: str
    total_seconds: int


class StatsOut(BaseModel):
    totals_by_label: list[LabelTotal]
    tasks_per_day: list[TasksPerDay]
    distribution: list[Distribution]
    daily_series: list[DailySeries]

File: backend/src/schemas/stats.ts
export interface LabelTotal {
  label: string;
  total_seconds: number;
}

export interface TasksPerDay {
  date: string;
  count: number;
}

export interface Distribution {
  label: string;
  percentage: number;
}

export interface DailySeries {
  date: string;
  total_seconds: number;
}

export interface StatsOut {
  totals_by_label: LabelTotal[];
  tasks_per_day: TasksPerDay[];
  distribution: Distribution[];
  daily_series: DailySeries[];
}